### PR TITLE
hacktoberfest-accepted

### PR DESCRIPTION
**Pull request by priyanka1111**
Added the condition to check if the array is already sorted. If array is already sorted in that case code should exit the loop after first pass only.

Added library for rand() function #include<stdlib,h>

### CODE REVIEW

**Comment by priyanka1111** [comment]
Hi, please review the changes done.

File: BubbleSort/main.cpp [hunk 1]
```diff
@@ -19,17 +20,23 @@ void printArray (int dizi[], int elemanSayisi) {
 
 void bubbleSort(int dizi[], int elemanSayisi) {
     int temp,i,j;
-    
+    bool is_swap = false;    //is_swap variable is set to true if any swap between any two elements occur
```
Comment (muhammedkarakul) [inline BubbleSort/main.cpp]: is_swap is snake case but I use camel case in program. If you can rename is_swap to isSwap, everythings ok.

File: BubbleSort/main.cpp [hunk 2]
```diff
@@ -7,6 +7,7 @@
 //
 
 #include <iostream>
+#include <stdlib.h> //for rand() function
```
Comment (muhammedkarakul) [inline BubbleSort/main.cpp]: I can't see rand() method usage. If this is unnecessary, please remove it.

File: BubbleSort/main.cpp [hunk 3]
```diff
@@ -19,17 +20,23 @@ void printArray (int dizi[], int elemanSayisi) {
...
                 temp = dizi[j];
                 dizi[j] = dizi[j+1];
                 dizi[j+1] = temp;
+                 is_swap = true;
             }
         }
+        if (is_swap == false)      //if no swap occurs implies all the elements are in sorted order. No need of conditions to loop to run further. 
+        {
+          break;   
+        }
```
Comment (muhammedkarakul) [inline BubbleSort/main.cpp]: I think code is very clear and readable. You can remove comment line or put it to above of if condition. Also "if" parenthesis usage is not same on the program. Put first parenthesis next of if condition and second one is below of if condition. This is not fatal issue but its important for standardization.

**Review by muhammedkarakul — COMMENTED** [review]
@priyanka1111 I'm sorry for the late reply. Thanks for your contribution. I review code and I gave some comments. If you follow the comments i can accept your pull request. Or explain your approach/reasons it may stay like this.

File: BubbleSort/main.cpp (hunk 1 above)
Comment (priyanka1111) [reply BubbleSort/main.cpp]: Done. Please check.

File: BubbleSort/main.cpp (hunk 3 above)
Comment (priyanka1111) [reply BubbleSort/main.cpp]: @muhammedkarakul Done please check.

File: BubbleSort/main.cpp (hunk 2 above)
Comment (priyanka1111) [reply BubbleSort/main.cpp]: https://github.com/muhammedkarakul/BubbleSortAlgorithm/blob/037712478c4b3313c86402eda66422efe33b2fd8/BubbleSort/main.cpp#L38

Comment (priyanka1111) [reply BubbleSort/main.cpp]: https://github.com/muhammedkarakul/BubbleSortAlgorithm/blob/037712478c4b3313c86402eda66422efe33b2fd8/BubbleSort/main.cpp#L40-L46

Comment (priyanka1111) [reply BubbleSort/main.cpp]: @muhammedkarakul  rand() function is used in Main.

File: BubbleSort/main.cpp (hunk 1 above)
Comment (muhammedkarakul) [reply BubbleSort/main.cpp]: Excellent! Only comment is not updated.

File: BubbleSort/main.cpp (hunk 2 above)
Comment (muhammedkarakul) [reply BubbleSort/main.cpp]: Ok sorry, my mistake. I was don't check full main.cpp code 👍

File: BubbleSort/main.cpp (hunk 3 above)
Comment (muhammedkarakul) [reply BubbleSort/main.cpp]: Done 👍

File: BubbleSort/main.cpp (hunk 1 above)
Comment (priyanka1111) [reply BubbleSort/main.cpp]: Comment removed. Please check.

Comment (muhammedkarakul) [reply BubbleSort/main.cpp]: 👌

**Comment by priyanka1111** [comment]
@muhammedkarakul Thank you. It would be grateful if you change the label of PR as **hacktoberfest-accepted**.

**Comment by muhammedkarakul** [comment]
Is it okey? @priyanka1111

**Comment by priyanka1111** [comment]
It would be grateful if u change label of PR as 'hacktoberfest-accepted'.

https://hacktoberfest-assets.nyc3.cdn.digitaloceanspaces.com/hacktoberfest-label-add-gif-min.gif

**Comment by priyanka1111** [comment]
@muhammedkarakul  .It would be grateful if u change label of PR as 'hacktoberfest-accepted'.

https://hacktoberfest-assets.nyc3.cdn.digitaloceanspaces.com/hacktoberfest-label-add-gif-min.gif

**Comment by muhammedkarakul** [comment]
@priyanka1111 i was too busy yesterday, sorry. I create "hacktoberfest-accepted" label and assign it to your PR.

**Comment by priyanka1111** [comment]
Thank you so much.

**Pull request merged**